process/elasticproc/tokens/sovereignIndexTokensHandler.go

```diff
@@ -0,0 +1,166 @@
+package tokens
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"math"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/elastic/go-elasticsearch/v7"
+	"github.com/multiversx/mx-chain-core-go/core"
+	"github.com/multiversx/mx-chain-core-go/data/esdt"
+
+	"github.com/multiversx/mx-chain-es-indexer-go/client"
+	"github.com/multiversx/mx-chain-es-indexer-go/client/logging"
+	"github.com/multiversx/mx-chain-es-indexer-go/data"
+	indexerdata "github.com/multiversx/mx-chain-es-indexer-go/process/dataindexer"
+	"github.com/multiversx/mx-chain-es-indexer-go/process/elasticproc"
+	"github.com/multiversx/mx-chain-es-indexer-go/process/elasticproc/converters"
+	"github.com/multiversx/mx-chain-es-indexer-go/process/elasticproc/factory"
+)
+
+type sovereignIndexTokensHandler struct {
+	indexingEnabled        bool
+	mainChainElasticClient elasticproc.DatabaseClientHandler
+	esdtPrefix             string
```

Comment:
What if you have multiple esdt prefixes from different chains?
For future requests from different chains we need to have a map[esdtPrefix]ElasticUrl

Reply:
I did only the sovereign indexer implementation